tests/test_migrations.py

```diff
@@ -32,76 +28,190 @@ def harvester():
+@pytest.mark.django_db
 @pytest.mark.cassandra
-def test_migrate_v2():
-    try:
-        RAW['doc'] = RAW['doc'].encode('utf-8')
-    except AttributeError:
-        RAW['doc'] = str(RAW['doc'])
-    DocumentModelOld.create(**RAW.attributes).save()
-    queryset = DocumentModel.objects(docID=RAW['docID'], source=RAW['source'])
-    assert len(queryset) == 0
-    tasks.migrate_to_source_partition(dry=False)
-    queryset = DocumentModel.objects(docID=RAW['docID'], source=RAW['source'])
-    assert len(queryset) == 1
+@pytest.mark.elasticsearch
+@pytest.mark.parametrize('canonical', ['postgres', 'cassandra'])
+@pytest.mark.parametrize('destination', ['postgres', 'cassandra', 'elasticsearch'])
+def test_cross_db(canonical, destination, monkeypatch, index='test'):
+
+    if canonical == destination:
+        return
+
+    monkeypatch.setattr('scrapi.settings.CANONICAL_PROCESSOR', canonical)
+
+    if destination != 'elasticsearch':
+        real_es = scrapi.processing.elasticsearch.ElasticsearchProcessor.manager.es
+        scrapi.processing.elasticsearch.es = mock.MagicMock()
+    else:
+        monkeypatch.setattr('scrapi.settings.ELASTIC_INDEX', 'test')
+
+    # Get the test documents into the caonical processor
+    canonical_processor = get_processor(canonical)
+    canonical_processor.process_raw(RAW)
+    canonical_processor.process_normalized(RAW, NORMALIZED)
+
+    destination_processor = get_processor(destination)
+
+    # Check to see canonical_processor is there, and destination is not
+    canonical_doc = canonical_processor.get(docID=RAW['docID'], source=RAW['source'])
+    assert canonical_doc
+
+    if destination != 'elasticsearch':
+        destination_doc = destination_processor.get(docID=RAW['docID'], source=RAW['source'])
+        assert not destination_doc
+    else:
+        destination_doc = destination_processor.get(index=index, source=RAW['source'])
+        normed = destination_doc.normalized
+        assert not normed
+
+    # Migrate from the canonical to the destination
+    tasks.migrate(cross_db, target_db=destination, dry=False, sources=['test'], index=index)
+
+    # Check to see if the document made it to the destinaton, and is still in the canonical
+    if destination != 'elasticsearch':
+        destination_doc = destination_processor.get(docID=RAW['docID'], source=RAW['source'])
+        assert destination_doc
+    else:
+        destination_doc = destination_processor.get(index=index, source=RAW['source'])
+        normed = destination_doc.normalized
+        assert normed
+
+    canonical_doc = canonical_processor.get(docID=RAW['docID'], source=RAW['source'])
+    assert canonical_doc
+
+    if destination != 'elasticsearch':
+        scrapi.processing.elasticsearch.es = real_es
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('destination', ['postgres', 'cassandra'])
+def test_no_doc_element_cross_db(destination, monkeypatch, index='test'):
+
+    if scrapi.settings.CANONICAL_PROCESSOR == destination:
```

Comment:
Using the cross_db migration for a database to itself didn't make sense to me, should it not be skipped? Like, cross_db postgres to postgres?

This just uses the canonical processor as the source, but I could cycle through everything (just didn't want to add more load)